mam/v2/sponge.c

```diff
@@ -148,6 +148,41 @@ void sponge_absorbn(isponge *s, trit_t c2, size_t n, trits_t *Xs) {
   } while (0 < m);
 }
 
+void sponge_absorbn_flex(isponge *s, trit_t c2, size_t n, trit_array_t Xs[]) {
+  size_t total_trits = 0;
+  for (size_t i = 0; i < n; ++i) {
+    total_trits += Xs[i].num_trits;
+  }
+  trit_t buffer[total_trits + n];
+  trits_t Xs_trits[n];
+  size_t trits_pos = 0;
+  for (size_t i = 0; i < n; ++i) {
+    flex_trits_to_trits(&buffer[trits_pos], Xs[i].num_trits, Xs[i].trits,
+                        Xs[i].num_trits, Xs[i].num_trits);
+    Xs_trits[i] = trits_from_rep(Xs[i].num_trits, &buffer[trits_pos]);
+    trits_pos += Xs[i].num_trits;
+  }
+
```

Comment:
Now that you have `Xs_trits` can't you just call `sponge_absorbn` for now ?

Reply:
The intention is to delete `sponge_absorbn` after we replace all usages